### DISCUSSION TITLE
json-based tetris presets

### DISCUSSION THREAD

**Discussion by chausch** [ppooll-dev/ppooll#68] [Ideas]
should tetris presets become json based as well?
not a pressing issue, but it would improve the standards-conformity of the project!

**Comment by klausfilip** [comment]
sure, but like you say, not pressing.

**Comment by the-steech** [comment]
i also think it might be a good idea to change how tetris "default" files are named & loaded.  live.ppooll relies on these tetris "default" files in order to set the correct dimensions for the containing bpatcher.  but maybe the size is accessible in another way that i'm not aware of :)

**Comment by klausfilip** [comment]
what do you mean? 
- the name misc/ppooll_factory_presets/actnameT/default is no good ?
- loaded? can be loaded at loadbang.
- 
is there a chance to write a ppooll.live wrapper, that goes for any act???

**Comment by klausfilip** [comment]
like eg.: let the actmaker look, if it is inside live and then load the wrapper?

**Comment by the-steech** [comment]
a wrapper for each act sounds interesting... it may require 2 bpatchers however, which off the top of my head seems redundant.  i will have some time this week to begin a branch where i can start adding the live.ppooll code, then we can all see how it works.  

the reason for renaming the default@ tho is because Max will try to "smartly" find files in Max paths, and I'm afraid somewhere down the line it will identify another acts "default@" and try to use it for the wrong act

**Comment by klausfilip** [comment]
why not load it with full path then?
Package:/ppooll/misc/ppooll_factory_presets/actnameT/default

**Comment by the-steech** [comment]
i have some javascript code that is grabbing the act window size from those files.  the problem with that is that if i look for a custom act that doesn't have a preset file, it will attempt to load "actnameT/default" (which doesn't exist).   Max then will look for any other file named "default" and attempt to use that.  

however, i'm realizing now quickest fix is for me to re-check the full path that it matches "actnameT/default", and if not, just return no size :)

**Comment by klausfilip** [comment]
ok! got it now. this js should give you notfound !

**Comment by klausfilip** [comment]
btw, creating a default tetris on save is possible with the [savebang] object.
but it would need to open tetris@ and then maybe ask if to always write the default tetris.

**Comment by klausfilip** [comment]
the tetris.json is there in main !!
- it was a long road again, but mainly deleting stuff and getting full advantage of dict.
- the nice part was that dict can be even managed with js - much more handy to write a dict there. 
- the default tetris now is written every time one saves an act !! (only factory acts at the moment - maybe write a default into ppooll_presets if its an unshared or contribution...
- amazing how much work i spent for a feature, that nobody really needs ;)

**Reply by the-steech** [reply]
amazing stuff, klaus!!

**Comment by chausch** [comment]
oh wowww 😮
just when i was not paying attention for a few days --- incredible work!!
sorry for opening up this topic, it was in no way my intention to fill up your work schedule 😬😅
🎉 🎉🎉

**Comment by klausfilip** [comment]
how to close a discussion? delete it?

**Reply by chausch** [reply]
you can lock the conversation – there's an option in the right sidebar, below the "unsubscribe" button!
